### ISSUE TITLE
[P3][dogfood] Canonical envelope says "<tool>: ok" verdict-blind — WARN/ERROR summary_line lies, inner recovery hint dropped

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#672]
## Summary

The canonical-envelope summary-line synthesizer says `"<tool>: ok"` **regardless of verdict**. On a WARN/ERROR/NOT_FOUND response the synthesized one-liner literally says **"ok"**, contradicting the verdict, and the inner tool's descriptive `summary_line` + recovery `next_step` are dropped.

Reproduces deterministically on the **live MCP server** (real consumer path, not `execute()`).

## Reproduction

`index action=status` against a project whose `.ast-cache` is empty/missing:

```jsonc
{
  "verdict": "WARN",
  "indexed": false,
  "hint": "Index missing or empty. Run the `index` tool with action=auto to build the cache.",
  "summary_line": "index: ok",          // ← lies: verdict is WARN
  "agent_summary": {
    "summary_line": "index: ok",        // ← lies
    "next_step": "",                    // ← inner tool's recovery hint was dropped
    "verdict": "WARN"
  }
}
```

The inner `CodeGraphStatusTool` correctly builds:
- `agent_summary.summary_line = "codegraph_status: index missing or empty"`
- `next_step = "Index missing or empty. Run the index tool with action=auto…"`

…but both are lost by the time the envelope reaches the agent. Healthy index is also affected (less harmfully): `verdict: INFO` + `"index: ok"` + `next_step: ""` — the inner's `"Index is healthy — proceed with nav or search tools"` next_step is also gone.

## Impact (P3, agent-UX)

An agent gating on `agent_summary.summary_line` / `next_step` (the documented place agents read) to decide *"is the index ready, what next?"* sees **"ok"** with no next step on the WARN path → proceeds to `nav`/`search` and gets phantom-empty results. The `verdict` field is correct, so a careful agent reading `verdict` is fine — that bounds severity to P3.

## Root cause (actionable)

`tree_sitter_analyzer/mcp/server_utils/error_recovery.py:428-429`:

```python
synthesized = (
    f"{tool_name}: {identifier_value}" if identifier_value else f"{tool_name}: ok"
)
```

The synthesizer is **verdict-blind** — it appends `": ok"` even when `response["verdict"]` is `WARN`/`ERROR`/`NOT_FOUND`. Two compounding issues:

1. **Verdict-blind "ok"**: when falling back to synthesis, the literal should reflect the verdict (e.g. `": index missing"` / `": warn"` / `": error"`), never an unconditional `"ok"`.
2. **Inner agent_summary lost through the facade**: the `index` facade delegates to `CodeGraphStatusTool` verbatim, yet the inner's rich `agent_summary.summary_line` + `next_step` don't survive to the boundary — forcing the synthesizer's step-2 fallback in the first place. The inner hint should be preserved (step-1 mirror should have used `"codegraph_status: index missing or empty"`).

## Suggested fix

- Make the synthesizer verdict-aware: only append `": ok"` when `verdict in (None, "INFO", "OK", "SUCCESS")`; otherwise derive the suffix from verdict/hint.
- Trace why the inner `agent_summary` is dropped before `_resolve_summary_line` runs for facade-routed status (so step-1 mirror fires instead of step-2 synthesis).

## RED-first test

`tests/unit/mcp/server_utils/` (or the envelope test that owns `_resolve_summary_line`): assert that a `{verdict: "WARN", ...}` response with no pre-set `summary_line` synthesizes a line that does **not** contain `"ok"` and that a verdict-appropriate token is present — exact-string pin, not substring-absence-only.

**Comment by aimasteracc** [comment]
Fixed in #678 (merged to develop). The canonical-envelope summary-line synthesizer now uses a success-like allowlist ({INFO, SAFE, OK, SUCCESS}) — those keep ': ok', while every other canonical verdict (WARN/ERROR/NOT_FOUND/CAUTION/REVIEW/UNSAFE, and any future one) synthesizes its own honest lowercased suffix, so a non-success response can never emit a lying '<tool>: ok'. (The secondary observation — facades stripping the inner tool's rich agent_summary before synthesis — remains a separate follow-up.)

**Comment by aimasteracc** [comment]
Reopening signal: PR #678 ("verdict-aware summary_line synthesis") missed the `health action=test_gap` path. On develop @ 55cac0f9, a bare `health action=test_gap` still returns the verdict-blind lie this issue described:

```json
{
  "verdict": "WARN",
  "coverage_pct": 14.9,
  "summary_line": "health: ok",
  "agent_summary": { "summary_line": "health: ok", "next_step": "", "verdict": "WARN" }
}
```

`verdict=WARN` but both `summary_line` and `agent_summary.summary_line` say `"health: ok"`. The verdict-aware synthesis from #678 isn't applied here — likely test_gap builds its own summary_line downstream of the synthesis point. (Separately, `next_step` is empty on a WARN with 103 gaps — same envelope, surfaced in the new test_gap scoping issue.)
